File: Projeto_8/models/sinalHandler.py
import signal
import sys
from models.persistencia import Persistencia

"""Trata os sinais"""
def tratarSinal(signal_received, frame):
    print("\n Sinal detectado! Salvando dados antes de encerrar...")
    dados_exemplo = [{"nome": "aluno exemplo", "matricula": "12345"}]
    Persistencia.salvar_dados("dados.txt", dados_exemplo)
    print("Dados salvos com sucesso! Encerrando o programa.")
    sys.exit(0)
  
signal.signal(signal.SIGINT, tratarSinal)  # CTRL+C
signal.signal(signal.SIGTERM, tratarSinal)  # Fechando o sistema

File: Projeto_8/PROJETO_8 - Williams Torres/src/SinalHandler.cpp
#include "SinalHandler.h"

void SinalHandler::inicializar() {
    std::signal(SIGINT, tratador);
    std::signal(SIGTERM, tratador);
    std::signal(SIGSEGV, tratador);
}

void SinalHandler::tratador(int sinal) {
    std::cout << "Sinal capturado: " << sinal << "\nSalvando dados e encerrando a aplicacao com seguranca." << std::endl;
    exit(1);
}